Model/Indexer/Data/Map/Update/Fetcher/Doofinder.php

```diff
@@ -146,8 +146,17 @@ private function getProductCollection(array $productIds, int $storeId, ?int $sto
          *         We override this behavior here.
          */
         $collection->setFlag('has_stock_status_filter', true);
-        $stockId = $stockId ?? $this->defaultStockProvider->getId();
-        $this->addStockDataToCollection->execute($collection, false, $stockId);
+
+        if ($this->moduleManager->isEnabled('Magento_InventoryCatalogApi')) {
```

Comment:
@Aitor-Corrales in this case i'm not sure if we should create a new helper or if it is fine doing the if else here. What do you think?

Reply:
I was wondering about it when i saw it in the code, but this conditional is only in a couple of functions so I do not think it's critical. In my opinion, it would be better placed in a private function within this very same helper but it's up to you